system/default_ad_api/src/compatibility/autoware_state.cpp

```diff
@@ -79,6 +71,21 @@ void AutowareStateNode::on_shutdown(
 
 void AutowareStateNode::on_timer()
 {
+  auto localization_msg = localization_sub_->takeData();
+  if (localization_msg) {
+    localization_state_ = *localization_sub_->takeData();
+  }
+
+  auto routing_msg = routing_sub_->takeData();
+  if (routing_msg) {
+    routing_state_ = *routing_sub_->takeData();
+  }
+
+  auto operation_mode_msg = operation_mode_sub_->takeData();
+  if (operation_mode_msg) {
+    operation_mode_state_ = *operation_mode_sub_->takeData();
+  }
+
```

Comment:
IMO
Calling `takeData()` twice for each msg is inefficient. Please reuse it.

Reply:
I agree! I'll make the changes.